File: app/src/main/java/me/ykrank/s1next/data/api/model/Favourite.java
package me.ykrank.s1next.data.api.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.Objects;

import org.apache.commons.lang3.StringEscapeUtils;

@SuppressWarnings("UnusedDeclaration")
@JsonIgnoreProperties(ignoreUnknown = true)
public final class Favourite {

    @JsonProperty("id")
    private String id;

    @JsonProperty("favid")
    private String favId;

    @JsonProperty("title")
    private String title;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        // unescape some basic XML entities
        this.title = StringEscapeUtils.unescapeXml(title);
    }

    public String getFavId() {
        return favId;
    }

    public void setFavId(String favId) {
        this.favId = favId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Favourite favourite = (Favourite) o;
        return Objects.equal(id, favourite.id) &&
                Objects.equal(favId, favourite.favId) &&
                Objects.equal(title, favourite.title);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id, favId, title);
    }
}

File: app/src/main/java/me/ykrank/s1next/data/api/model/Favourite.kt
package me.ykrank.s1next.data.api.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.github.ykrank.androidtools.ui.adapter.StableIdModel
import me.ykrank.s1next.util.HtmlUtils

@JsonIgnoreProperties(ignoreUnknown = true)
class Favourite : StableIdModel {

    @JsonProperty("id")
    var id: String? = null

    @JsonProperty("favid")
    var favId: String? = null

    @JsonProperty("title")
    // unescape some basic XML entities
    var title: String? = null
        set(title) {
            field = HtmlUtils.unescapeHtml(title)
        }

    override val stableId: Long
        get() = favId?.toLongOrNull() ?: 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Favourite

        if (id != other.id) return false
        if (favId != other.favId) return false
        if (title != other.title) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (favId?.hashCode() ?: 0)
        result = 31 * result + (title?.hashCode() ?: 0)
        return result
    }

}
